vendor-extra/JatsContentBundle/src/EventListener/CreateView/FrontSubjectGroupContentHeaderListener.php

```diff
@@ -2,37 +2,37 @@
 
 declare(strict_types=1);
 
-namespace Libero\JatsContentBundle\ViewConverter;
+namespace Libero\JatsContentBundle\EventListener\CreateView;
 
 use DOMNodeList;
 use FluentDOM\DOM\Element;
+use Libero\ViewsBundle\Event\CreateViewEvent;
+use Libero\ViewsBundle\Views\ContextAwareTranslation;
 use Libero\ViewsBundle\Views\ConvertsLists;
-use Libero\ViewsBundle\Views\SimplifiedVisitor;
-use Libero\ViewsBundle\Views\TranslatingVisitor;
+use Libero\ViewsBundle\Views\SimplifiedViewConverterListener;
 use Libero\ViewsBundle\Views\View;
 use Libero\ViewsBundle\Views\ViewConverter;
-use Libero\ViewsBundle\Views\ViewConverterVisitor;
 use Symfony\Contracts\Translation\TranslatorInterface;
 use function array_map;
 use function count;
 
-final class FrontSubjectGroupContentHeaderVisitor implements ViewConverterVisitor
+final class FrontSubjectGroupContentHeaderListener
 {
+    use ContextAwareTranslation;
     use ConvertsLists;
-    use SimplifiedVisitor;
-    use TranslatingVisitor;
-
-    private $converter;
-    private $translator;
+    use SimplifiedViewConverterListener;
 
     public function __construct(ViewConverter $converter, TranslatorInterface $translator)
     {
         $this->converter = $converter;
         $this->translator = $translator;
     }
 
-    protected function doVisit(Element $object, View $view) : View
+    protected function handle(CreateViewEvent $event) : View
```

Comment:
Easier to pass the `Element` and `View` directly, but won't be able to stop events. Maybe the `Event` should be a third argument, and (mostly) not needed?

Reply:
Works, but PHPStan (correctly) complains `invoked with 3 parameters, 2 required`. Unless it is typehinted and left unused.

Reply:
Maybe if you want the event it shouldn't be using the 'simplified' trait. The template etc validation could be extracted into a separate one.

Reply:
dabae39